src/makeInfinityAdder.js

```diff
@@ -6,7 +6,26 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
-}
+  let sum = 0;
+
+  function adder(num) {
+    if (typeof num === 'number') {
+      sum += num;
+
+      return adder;
+    } else {
+      const result = sum;
 
+      sum = 0;
+
+      return result;
+    }
+  }
+
+  if (arguments.length > 0) {
+    return adder(...arguments);
+  } else {
+    return adder;
```

Comment:
```suggestion

return adder;
```
looks like redundant check